### PR TITLE
Add a documentation for op decomposition

### PR DESCRIPTION
**Pull request by jserbedzijaTT**

### CODE REVIEW

**Review by sdjordjevicTT — APPROVED** [review]
Thanks @jserbedzijaTT for documenting this! :)

File: docs/src/decomposing-an-op-in-ttir.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+For the `Index` operation, we use the C++ conversion pattern because it involves changing the Op’s input types from integers to arrays, which TableGen lacks flexibility for.
+
+```
+{{#include ../../../include/ttmlir/Dialect/TTNN/IR/TTNNOps.td:adding_an_op_index_ttir}}
+```
+
+The `matchAndRewrite` method from `OpConversionPattern` is implemented to replace the matched Op with the newly created Op. Since decomposition is implemented as a conversion pass, `OpAdaptor` is used to access the attributes of the original Op in their converted types. Finally, we instantiate the new Op and call the `replaceOp` method on `ConversionPatternRewriter` to replace the original Op.
+
+#### `Tablegen conversion pattern`
+TODO
```
Comment (nsmithtt) [inline docs/src/decomposing-an-op-in-ttir.md:44]: Can you file an issue?

**Review by nsmithtt — APPROVED** [review]
Awesome, thanks for adding these docs!

File: docs/src/decomposing-an-op-in-ttir.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+- [Decomposing an Op in TTIR](#decomposing-an-op-in-ttir)
+  - [1. Define the Op in the TTIR frontend dialect](#1-define-the-op-in-the-ttir-frontend-dialect)
+  - [2. Create a conversion pattern](#2-create-a-conversion-pattern)
+      - [`C++ conversion pattern`](#c-conversion-pattern)
+      - [`Tablegen conversion pattern`](#tablegen-conversion-pattern)
+  - [3. Register the created conversion pattern](#3-register-the-created-conversion-pattern)
+
+## 1. Define the Op in the TTIR frontend dialect
+
+The more information regarding this step can be found here: [Define the Op in the TTIR frontend dialect](./adding-an-op.md#1-define-the-op-in-the-ttir-frontend-dialect).
```
Comment (svuckovicTT) [inline docs/src/decomposing-an-op-in-ttir.md:15]: Remove the `The`.

File: docs/src/decomposing-an-op-in-ttir.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
+# Decomposing an Op in TTIR
+
+This guide explains how to add and decompose a new operation end-to-end in tt-mlir. We’ll focus on adding an `Index` operation, which will be decomposed into the `Slice` operation. The decomposition is implemented as a conversion pass in MLIR since it allows us to mark operations or dialects as legal or illegal, type conversion...
```
Comment (svuckovicTT) [inline docs/src/decomposing-an-op-in-ttir.md:3]: > This guide explains how to add and decompose a new operation end-to-end

I understood that this guide was going to explain how adding an op end-to-end should be done in tt-mlir. Maybe a little tweaking of the sentence would make it more accurate, something like `This guide explains how to add and decompose a new op in TTIR dialect.`.

**Review by svuckovicTT — APPROVED** [review]
Awesome work Jovan!

File: docs/src/decomposing-an-op-in-ttir.md (hunk 3 above)
Comment (jserbedzijaTT) [reply docs/src/decomposing-an-op-in-ttir.md:3]: Yeah, I was not really precise in this explanation. I will refactor it, thanks for the heads up.

**Pull request merged**